src/jsonpath/parser.c

```diff
@@ -333,19 +376,23 @@ bool build_parse_tree(lex_token lex_tok[PARSE_BUF_LEN], char lex_tok_values[][PA
         }
 
         break;
-      case LEX_LITERAL_BOOL:
-        cur = ast_alloc_node(cur, AST_LITERAL_BOOL);
-
-        if (jp_str_cpy(cur->data.d_literal.value, PARSE_BUF_LEN, lex_tok_values[*lex_idx],
-                       strlen(lex_tok_values[*lex_idx])) > 0) {
-          zend_throw_exception_ex(spl_ce_RuntimeException, 0, "Buffer size exceeded");
+      case LEX_LITERAL_NUMERIC:
+        cur = ast_alloc_node(cur, AST_DOUBLE);
+        if (!make_numeric_node(cur, lex_tok_values[*lex_idx], strlen(lex_tok_values[*lex_idx]))) {
+          zend_throw_exception_ex(spl_ce_RuntimeException, 0, "Unable to parse numeric.");
           return false;
         }
+        break;
+      case LEX_LITERAL_BOOL:
+        cur = ast_alloc_node(cur, AST_BOOL);
 
-        if (strcmp(cur->data.d_literal.value, "true") == 0) {
-          jp_str_cpy(cur->data.d_literal.value, PARSE_BUF_LEN, "JP_LITERAL_TRUE", 15);
-        } else if (strcmp(cur->data.d_literal.value, "false") == 0) {
-          jp_str_cpy(cur->data.d_literal.value, PARSE_BUF_LEN, "JP_LITERAL_FALSE", 16);
```

Comment:
Happy to see this hack of mine gone!

Reply:
Ah hah, I was wondering where that code came from. I assumed I put it in there years ago without looking at the commit history.